test/base.test.js

```diff
@@ -669,3 +669,151 @@ test('skip the cache if operation is Mutation', async ({ equal, same, teardown }
   equal(skipCount, 0)
   equal(hitCount, 0)
 })
+
+test('using all option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+
+  try {
+    await app.register(cache, {
+      all: 'true'
+    })
+  } catch (error) {
+    t.same(error, new Error('all must be an boolean'))
+  }
+})
+
+test('using ttl option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+  try {
+    await app.register(cache, {
+      ttl: '10'
+    })
+  } catch (error) {
+    t.same(error, new Error('ttl must be a number'))
+  }
+})
+
+test('using cacheSize option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+  try {
+    await app.register(cache, {
+      cacheSize: '1024'
+    })
+  } catch (error) {
+    t.same(error, new Error('cacheSize must be a number'))
+  }
+})
+
+test('using onHit option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+  try {
+    await app.register(cache, {
+      onHit: 'not a function'
+    })
+  } catch (error) {
+    t.same(error, new Error('onHit must be a function'))
+  }
+})
+
+test('using onMiss option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+  try {
+    await app.register(cache, {
+      onMiss: 'not a function'
+    })
+  } catch (error) {
+    t.same(error, new Error('onMiss must be a function'))
+  }
+})
+
+test('using onSkip option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+  try {
+    await app.register(cache, {
+      onSkip: 'not a function'
+    })
+  } catch (error) {
+    t.same(error, new Error('onSkip must be a function'))
+  }
+})
+
+test('using policy option as string', async (t) => {
+  t.plan(1)
+  const app = fastify()
+  app.register(mercurius)
+  try {
+    await app.register(cache, {
+      policy: 'not an object',
+      all: false
+    })
+  } catch (error) {
+    t.same(error, new Error('policy must be an object'))
+  }
+})
+
+test('Unmatched schema for Query', async ({ same, teardown }) => {
+  const app = fastify()
+  teardown(app.close.bind(app))
+
+  const schema = `
+      type Query {
+        add(x: Int, y: Int): Int
+      }
+    `
+
+  const resolvers = {
+    Query: {
+      async add (_, { x, y }) {
+        await immediate()
+        return x + y
+      }
+    }
+  }
+
+  app.register(mercurius, {
+    schema,
+    resolvers
+  })
+
+  app.register(cache, {
+    policy: {
+      Query: {
+        add: true,
+        foo: 'bar'
+      }
+    }
+  })
```

Comment:
I would simplify this test case. It should not be necessary to call `inject` to get the error.

```suggestion
try {
  app.register(cache, {
    policy: {
      Query: {
        add: true,
        foo: 'bar'
      }
    }
  })
  t.fail('should not load correctly')
} catch (err) {
  same(error, new Error('Query does not match schema'))
}
```

Reply:
I also thought that was the case. this was actually where I was stuck before. I was trying without inject but that does not cover the the if condition

<img width="595" alt="Screenshot 2021-11-29 at 15 00 44" src="https://user-images.githubusercontent.com/17081313/143891149-1a75e067-cc3f-4e4c-9ecc-2384c6fb19f1.png">

Reply:
Does it worth adding both tests?

- one to check that the loading is not performed (so we are sure that the server is not listening)
- one to complete the coverage?